components/03-modules/modal/modal.config.js

```diff
@@ -1,16 +1,57 @@
 module.exports = {
   context: {
-    class: '',
-    id: 'modal-1',
+    modalContent: {
+      class: '',
+    },
+    modal: {
+      class: '',
+      id: 'myDialog',
+      // read README.md when implementing modal
+      attributes: 'aria-labelledby="myTitle" aria-describedby="myDesc"'
+    },
     trigger: true,
-    content: 'button',
+    modalTrigger: {
+      buttonModalTrigger: {
+        tag: 'button',
+        class: 'modal-trigger',
+        text: 'Modal trigger button',
+        attributes: 'data-modaltrigger="myDialog" type="button"'
+      }
+    },
+    main: {
+      html: '<h3 id="myTitle">Save "untitled" document?</h3><div id="myDesc">You have made changes to "untitled.txt" that have not been saved. What do you want to do?</div>'
+    },
     buttonClose: {
       tag: 'button',
       text: '',
-      class: 'button--icon modal__close-button',
+      class: 'button--icon modal__js-close-button modal__close-button',
       iconId: 'close',
       iconClass: 'button__icon modal__close-button-icon',
       attributes: 'type="button" aria-label="close modal button, click to close the modal"'
-    }
-  }
+    },
+    script: true
+  },
+  variants: [
+    {
+      name: 'header-footer',
+      context: {
+        header: {
+          html: '<h3 id="myTitle">Save "untitled" document?</h3>',
+        },
+        main: {
+          html: '<div id="myDesc">You have made changes to "untitled.txt" that have not been saved. What do you want to do?</div>'
+        },
+        footer: {
+          html: 'Some text here or via component',
+          content: 'button',
+          contentContext: {
+            text: 'Close',
+            tag: 'button',
+            class: 'button modal__js-close-button',
+            attributes: 'type="button" aria-label="Close"'
```

Comment:
When button has text, `aria-label` is useless b/c screen reader will read the text inside, it good to use it when inside is the icon and when you want to set other text for screen readers that is visible in the button

Reply:
sure. done